tests/codegen-llvm/deduced-param-attrs.rs

```diff
@@ -47,14 +51,34 @@ pub fn use_big_cell_container(big_cell_container: BigCellContainer) {
 // Make sure that we don't mistakenly mark a big struct as `readonly` when passed through a generic
 // type parameter if it contains UnsafeCell.
 //
-// CHECK-NOT: @use_something({{.*}} readonly {{.*}} %something)
+// CHECK: @use_something(
+// CHECK-NOT: readonly
+// CHECK-SAME: %something)
 #[no_mangle]
 #[inline(never)]
 pub fn use_something<T>(something: T) {
     hint::black_box(&something);
 }
 
+// Make sure that we still mark a big `Freeze` struct as `readonly` when passed through a generic
+// type parameter.
+//
+// CHECK: @use_something_freeze(
+// CHECK-SAME: readonly
+// CHECK-SAME: %x)
+#[no_mangle]
+#[inline(never)]
+pub fn use_something_freeze<T>(x: T) {
+    // `Drop` counts as a mutable use.
+    std::mem::forget(x)
+}
```

Comment:
It would be possible to use similar approach with respect to drop: check if type needs dropping during ABI adjustment if there are any drops of a local.

This is one situation where having a drop terminator instead of a lowered `drop_in_place` would be more convenient to work with (referencing the recent discussion on zulip cc @scottmcm).